src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorSpan.cs

```diff
@@ -690,37 +690,70 @@ public TensorSpan<T> Slice(params scoped ReadOnlySpan<NRange> ranges)
             if (ranges.Length != Lengths.Length)
                 ThrowHelper.ThrowIndexOutOfRangeException();
 
+            TensorSpan<T> toReturn;
             scoped Span<nint> lengths;
             scoped Span<nint> offsets;
+            nint[]? lengthsArray;
+            nint[]? offsetsArray;
             if (Rank > TensorShape.MaxInlineRank)
             {
-                lengths = stackalloc nint[Rank];
-                offsets = stackalloc nint[Rank];
+                lengthsArray = ArrayPool<nint>.Shared.Rent(Rank);
+                lengths = lengthsArray;
+                lengths = lengths.Slice(0, Rank);
+                lengthsArray = null;
+
+                offsetsArray = ArrayPool<nint>.Shared.Rent(Rank);
+                offsets = offsetsArray;
+                offsets = offsets.Slice(0, Rank);
+                offsetsArray = null;
```

Comment:
Same comments/questions as on the readonly variant.

Reply:
Fixed.